pkg/controller/acmeorders/sync_test.go

```diff
@@ -514,6 +542,53 @@ rUCGwbCUDI0mxadJ3Bz4WxR6fyNpBK2yAinWEsikxqEt
 				},
 			},
 		},
+		"call FinalizeOrder and update the order state to 'errored' if finalize fails with a 4xx ACME error": {
```

Comment:
may be worth adding a test case for the instance where it returns an error because the order is already finalized, to ensure it proceeds gracefully (I am not 100% sure what response code/body is returned in this case either, so may need some digging into)

Reply:
It will be 403 (see [here](https://github.com/letsencrypt/boulder/blob/main/wfe2/wfe.go#L2280-L2288) and [here](https://github.com/letsencrypt/boulder/blob/6184ad5365ebdf968ecaa86ef30bc58a73a46440/probs/probs.go#L324-L330) so should fall under this test case